hoomd/mpcd/CollisionMethod.cc

```diff
@@ -48,15 +48,272 @@ void mpcd::CollisionMethod::collide(uint64_t timestep)
+            m_angmom_accum.swap(angmom_accum);
+            }
+        beginRigidBodyCollision(timestep);
+        }
+
+    // apply collisions
     rule(timestep);
+
+    // apply collisions to rigid bodies
+    if (m_embed_group)
+        {
+        finishRigidBodyCollision(timestep);
+        }
+    }
+
+void mpcd::CollisionMethod::checkCollisionWarnings(uint64_t timestep)
+    {
+    if (m_checked_collision_warnings)
+        {
+        return;
+        }
+
+    if (m_embed_group)
+        {
+        const unsigned int num_group = m_embed_group->getNumMembers();
+        ArrayHandle<unsigned int> h_embed_group(m_embed_group->getIndexArray(),
+                                                access_location::host,
+                                                access_mode::read);
+        ArrayHandle<Scalar4> h_vel_embed(m_pdata->getVelocities(),
+                                         access_location::host,
+                                         access_mode::read);
+        ArrayHandle<unsigned int> h_body_embed(m_pdata->getBodies(),
+                                               access_location::host,
+                                               access_mode::read);
+        ArrayHandle<unsigned int> h_rtag_embed(m_pdata->getRTags(),
+                                               access_location::host,
+                                               access_mode::read);
+
+        // check if some of the masses are less or equal to 0
+        bool invalid_mass = false;
+        bool central_interacting = false;
+        for (unsigned int idx = 0; idx < num_group && !(invalid_mass && central_interacting); ++idx)
+            {
+            // get the index from the embedded group
+            const unsigned int particle_index = h_embed_group.data[idx];
+
+            // check mass
+            const Scalar4 vel_mass = h_vel_embed.data[particle_index];
+            const Scalar mass = vel_mass.w;
+            if (mass <= Scalar(0))
+                {
+                invalid_mass = true;
+                }
+
+            // check if particle is central particle in rigid body
+            const unsigned int central_tag = h_body_embed.data[particle_index];
+            if (central_tag < MIN_FLOPPY)
+                {
+                const unsigned int central_idx = h_rtag_embed.data[central_tag];
+                if (particle_index == central_idx)
+                    {
+                    central_interacting = true;
+                    }
+                }
+            }
+
+#ifdef ENABLE_MPI
+        // MPI_Reduce invalid_mass
+        unsigned int invalid_mass_int = (unsigned int)invalid_mass;
+        unsigned int invalid_mass_reduced;
+        MPI_Reduce(&invalid_mass_int,
+                   &invalid_mass_reduced,
+                   1,
+                   MPI_UNSIGNED,
+                   MPI_LOR,
+                   0,
+                   m_exec_conf->getMPICommunicator());
```

Comment:
This should probably be `Allreduce` so all ranks know the result, and you can use `MPI_IN_PLACE` and the `MPI_CXX_BOOL` type to simplify things.
```suggestion
        MPI_Allreduce(MPI_IN_PLACE,
                   &invalid_mass,
                   1,
                   MPI_CXX_BOOL,
                   MPI_LOR,
                   m_exec_conf->getMPICommunicator());
```
Please apply the same changes to the other one, and also wrap this in a check on the number of ranks being > 1.